torch/compiler/config.py

```diff
@@ -0,0 +1,114 @@
+# grouping clear.
+
+cache: bool = True
+"""When ``True`` (default), enable all (non-experimental) caches supported by
+the compiler, persisting compilation products across multiple invocations of
+the compiler.  When ``False``, do not enable caches by default (however, if an
+individual cache is explicitly enabled, we will respect those settings.)  By
+default, caches are written local filesystem at
+``/tmp/torchinductor_$USERNAME``.
+
+At time of writing (PyTorch 2.5), the following caches are enabled by default:
+
+* Triton cache, which caches cubins generated by the Triton compiler
+  (technically, this cache is implemented by the Triton compiler and not by the
+  PyTorch compiler, but this configuration option does control whether or not
+  the Triton cache is used.)
+
+* Inductor autotune cache, which caches autotuning results from Inductor
+  compilation (this includes tuning for both Inductor generated kernels as
+  well as matrix multiply algorithm selection, among others).  This cache is
+  controlled by :data:`torch._inductor.config.autotune_local_cache`,
+  :data:`torch._inductor.config.autotune_remote_cache`
+  (:envvar:`TORCHINDUCTOR_AUTOTUNE_REMOTE_CACHE`) and
+  :data`torch._inductor.config.bundled_autotune_remote_cache`
+  (:envvar:`TORCHINDUCTOR_BUNDLED_AUTOTUNE_REMOTE_CACHE`).
+
+* Inductor FX cache, which caches the Inductor phase of the compiler.
+  This cache currently does not subsume the Triton cache, as it only
+  caches Python source code that must then be compiled, but as of PyTorch 2.5
+  we are in the process of changing this.  This cache is controlled by
+  :data:`torch._inductor.config.fx_graph_cache`
+  (:envvar:`TORCHINDUCTOR_FX_GRAPH_CACHE`) and
+  :data:`torch._inductor.config.fx_graph_remote_cache`
+  (:envvar:`TORCHINDUCTOR_FX_GRAPH_REMOTE_CACHE`)
+
+The following experimental caches are not currently enabled by default:
+
+* AOTAutograd cache, which caches the AOTAutograd phase of the compiler.  This
+  cache subsumes the Inductor FX cache.  This cache is controlled by
+  :data:`torch._functorch.config.enable_autograd_cache`
+  (:envvar:`TORCHINDUCTOR_AUTOGRAD_CACHE`) and
+  :data:`torch._functorch.config.enable_remote_autograd_cache`
+  (:envvar:`TORCHINDUCTOR_AUTOGRAD_REMOTE_CACHE`).
+
+* Automatic dynamic profile-guided optimization (PGO) cache, which caches
+  automatic dynamic determinations for inputs to the graph.  This cache is
+  not yet landed to PyTorch nightlies yet.
+
+We reserve the right to change what caches are enabled by default.  Our
+contract to you is that caches we enable by default should never introduce a
+correctness problem.  You opt-out of the default set of caches by setting
+``TORCH_COMPILE_CACHE=0`` (this still alows for individual caches to be
+enabled; if you need to guarantee caches are disabled even if user code has
+explicitly enabled them by modifying :data:`torch.compiler.config`, use
+``TORCH_COMPILE_FORCE_CACHE=0``).
```

Comment:
Critical part 2: Environment variables do not override changes to config values from user code. However, a `FORCE` envvar overrides everything.